src/shared/watchdog.c

```diff
@@ -19,26 +19,34 @@ static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
+/* Starting from kernel version 4.5, the maximum allowable watchdog timeout is
+ * UINT_MAX/1000U seconds (since internal calculations are done in milliseconds).
+ */
+#define WATCHDOG_TIMEOUT_MAX_SEC (UINT_MAX/1000U)
```

Comment:
please consider writing this as CONST_MIN(UINT_MAX/1000U, INT_MAX)

see my earlier comments

Reply:
I just responded with my thoughts above, let me know what you think.